Marlin/Marlin_main.cpp

```diff
@@ -9363,14 +9363,19 @@ void set_current_from_steppers_for_axis(const AxisEnum axis) {
       // For non-interpolated delta calculate every segment
       for (uint16_t s = segments + 1; --s;) {
         DELTA_NEXT(segment_distance[i]);
-        planner.buffer_line_kinematic(DELTA_VAR, _feedrate_mm_s, active_extruder);
+        DELTA_IK();
+        ADJUST_DELTA(DELTA_VAR); // Adjust Z if bed leveling is enabled
+        planner.buffer_line(delta[A_AXIS], delta[B_AXIS], delta[C_AXIS], DELTA_VAR[E_AXIS], _feedrate_mm_s, active_extruder);
```

Comment:
The `planner.buffer_line_kinematic` function does handle the leveling, so that function will do the right thing.

However, it does look like this fixes something. The `USE_RAW_KINEMATICS` feature needs to use the macros to do kinematics in raw form. So this might be correct for that reason.